custom_components/tahoma/state_sensor.py

```diff
@@ -0,0 +1,151 @@
+    value: Callable[[Any], Any] = lambda val: val
+    device_class: Optional[str] = None
+    default_enabled: bool = True
+
+
+supported_states = {
+    "core:BatteryState": StateDescription(
+        key="core:BatteryState",
+        name="Battery",
+        unit=PERCENTAGE,
+        device_class=sensor.DEVICE_CLASS_BATTERY,
+    ),
+    "core:RSSILevelState": StateDescription(
+        key="core:RSSILevelState",
+        name="RSSI Level",
+        value=lambda value: round(value),
+        unit=SIGNAL_STRENGTH_DECIBELS,
+        device_class=sensor.DEVICE_CLASS_SIGNAL_STRENGTH,
+    ),
+    "core:ExpectedNumberOfShowerState": StateDescription(
+        key="core:ExpectedNumberOfShowerState",
+        name="Expected Number Of Shower",
+        icon="mdi:shower-head",
+        value=lambda value: round(value),
+    ),
+    "core:NumberOfShowerRemainingState": StateDescription(
+        key="core:NumberOfShowerRemainingState",
+        name="Number of Shower Remaining",
+        icon="mdi:shower-head",
+        value=lambda value: round(value),
+    ),
+    # V40 is measured in litres (L) and shows the amount of warm (mixed) water with a temperature of 40 C, which can be drained from a switched off electric water heater.
+    "core:V40WaterVolumeEstimationState": StateDescription(
+        key="core:V40WaterVolumeEstimationState",
+        name="Water Volume Estimation at 40 °C",
+        icon="mdi:water",
+        value=lambda value: round(value),
+        unit=VOLUME_LITERS,
+        default_enabled=False,
+    ),
+    "core:WaterConsumptionState": StateDescription(
+        key="core:WaterConsumptionState",
+        name="Water Consumption",
+        icon="mdi:water",
+        value=lambda value: round(value),
+        unit=VOLUME_LITERS,
+    ),
+    "io:OutletEngineState": StateDescription(
+        key="io:OutletEngineState",
+        name="Outlet Engine",
+        icon="mdi:fan-chevron-down",
+        value=lambda value: round(value),
+        unit=VOLUME_LITERS,
+    ),
+    "io:InletEngineState": StateDescription(
+        key="io:InletEngineState",
+        name="Inlet Engine",
+        icon="mdi:fan-chevron-up",
+        value=lambda value: round(value),
+        unit=VOLUME_FLOW_RATE_CUBIC_METERS_PER_HOUR,
+    ),
+    "hlrrwifi:RoomTemperatureState": StateDescription(
+        key="hlrrwifi:RoomTemperatureState",
+        name="Room Temperature",
+        value=lambda value: round(value),
+        device_class=sensor.DEVICE_CLASS_TEMPERATURE,
+    ),
+    "io:MiddleWaterTemperatureState": StateDescription(
+        key="io:MiddleWaterTemperatureState",
+        name="Middle Water Temperature",
+        value=lambda value: round(value),
+        device_class=sensor.DEVICE_CLASS_TEMPERATURE,
+    ),
+}
+
+
+class TahomaStateSensor(TahomaEntity, Entity):
+    """Representation of a TaHoma Sensor, based on a secondary device."""
+
+    def __init__(
+        self,
+        device_url: str,
```

Comment:
I would add the key as a parameter and remove it from the state_description to avoid the duplicate data. https://github.com/iMicknl/ha-tahoma/pull/425/files#r615452018